test/e2e/upg_e2e.go

```diff
@@ -526,6 +529,132 @@ var _ = ginkgo.Describe("Binapi", func() {
+				gomega.Expect(msg.Teid).To(gomega.BeEquivalentTo(gtpuEndpoint.Teid))
+			}
+			gomega.Expect(found).To(gomega.BeTrue())
+			gtpuEndpoint.IsAdd = 0
+			gtpuEndpointReply = &upf.UpfGtpuEndpointAddDelReply{}
+			err = f.VPP.ApiChannel.SendRequest(gtpuEndpoint).ReceiveReply(gtpuEndpointReply)
+			gomega.Expect(err).To(gomega.BeNil())
+			reqCtx = f.VPP.ApiChannel.SendMultiRequest(&upf.UpfGtpuEndpointDump{})
+			found = false
+			for {
+				msg := &upf.UpfGtpuEndpointDetails{}
+				stop, err := reqCtx.ReceiveReply(msg)
+				gomega.Expect(err).To(gomega.BeNil())
+				if stop {
+					break
+				}
+				if msg.IP4 != gtpuEndpoint.IP4 {
+					continue
+				}
+				found = true
+			}
+			gomega.Expect(found).To(gomega.BeFalse())
+		})
+	})
+	ginkgo.Context("for PFCP endpoint", func() {
+		f := framework.NewDefaultFramework(framework.UPGModeTDF, framework.UPGIPModeV4)
+		ginkgo.It("lists and removes the PFCP endpoint", func() {
+
+			// It appears to be a little tricky to add new PFCP endpoint
+			// So far we will just check if it can list and remove PFCP endpoint
+			// Adding of endpoint will be checked during the change of configuration mechanism
+			pfcpEndpoint := &upf.UpfPfcpEndpointAddDel{
+				IsAdd: 0,
+				TableID: 0,
+			}
+			ipAddr, er := ip_types.ParseAddress("10.0.0.2")
+			gomega.Expect(er).To(gomega.BeNil())
+			pfcpEndpoint.IP = ipAddr
+
+			reqCtx := f.VPP.ApiChannel.SendMultiRequest(&upf.UpfPfcpEndpointDump{})
+			var found = false
+
+			for {
+				msg := &upf.UpfPfcpEndpointDetails{}
+				stop, err := reqCtx.ReceiveReply(msg)
+				gomega.Expect(err).To(gomega.BeNil())
+				if stop {
+					break
+				}
+				if msg.IP != pfcpEndpoint.IP {
+					continue
+				}
+				found = true
+				gomega.Expect(msg.FibTable).To(gomega.BeEquivalentTo(0))
+			}
+			gomega.Expect(found).To(gomega.BeTrue())
+
+			pfcpEndpointReply := &upf.UpfPfcpEndpointAddDelReply{}
+			err := f.VPP.ApiChannel.SendRequest(pfcpEndpoint).ReceiveReply(pfcpEndpointReply)
+			gomega.Expect(err).To(gomega.BeNil())
+			reqCtx = f.VPP.ApiChannel.SendMultiRequest(&upf.UpfPfcpEndpointDump{})
+			found = false
+			for {
+				msg := &upf.UpfPfcpEndpointDetails{}
+				stop, err := reqCtx.ReceiveReply(msg)
+				gomega.Expect(err).To(gomega.BeNil())
+				if stop {
+					break
+				}
+				if msg.IP != pfcpEndpoint.IP {
+					continue
+				}
+				found = true
+			}
+			gomega.Expect(found).To(gomega.BeFalse())
+		})
+	})
+	ginkgo.Context("for PFCP Session Server", func() {
+		f := framework.NewDefaultFramework(framework.UPGModeTDF, framework.UPGIPModeV4)
+		ginkgo.It("Configures PFCP Server", func() {
+
+			sessionServerCfg := &upf.UpfPfcpServerSet{
+				FifoSize: 512, // KB
+				SegmentSize: 512, // MB
+			}
+			reply := &upf.UpfPfcpServerSetReply{}
+
+			err := f.VPP.ApiChannel.SendRequest(sessionServerCfg).ReceiveReply(reply)
+			// TODO: This will return error, need to change configuration mechanism of UPGMode to binapi
+			gomega.Expect(err).NotTo(gomega.BeNil())
```

Comment:
It would be nice to have an API for retrieving the settings back. But even if for some reason we don't do it (even though better we do), it would be nice to run some `f.VPP.Ctl(...)` here and check the result for some substrings

Reply:
done